crates/compilers/src/compilers/resolc/compiler.rs

```diff
@@ -0,0 +1,196 @@
+use crate::{
+    error::{Result, SolcError},
+    resolver::parse::SolData,
+    solc::{Solc, SolcCompiler, SolcSettings},
+    Compiler, CompilerVersion,
+};
+use foundry_compilers_artifacts::{resolc::ResolcCompilerOutput, Contract, Error, SolcLanguage};
+use itertools::Itertools;
+use semver::Version;
+use serde::Serialize;
+use std::{
+    io,
+    io::Write,
+    path::{Path, PathBuf},
+    process::{Command, Output, Stdio},
+    str::FromStr,
+};
+
+use super::{ResolcInput, ResolcVersionedInput};
+
+#[derive(Clone, Debug)]
+pub struct Resolc {
+    pub resolc: PathBuf,
+    pub resolc_version: Version,
+    pub solc: SolcCompiler,
+}
+
+impl Compiler for Resolc {
+    type CompilerContract = Contract;
+    type Input = ResolcVersionedInput;
+    type CompilationError = Error;
+    type ParsedSource = SolData;
+    type Settings = SolcSettings;
+    type Language = SolcLanguage;
+
+    /// Instead of using specific sols version we are going to autodetect
+    /// Installed versions
+    fn available_versions(&self, language: &SolcLanguage) -> Vec<CompilerVersion> {
+        self.solc.available_versions(language)
+    }
+
+    fn compile(
+        &self,
+        input: &Self::Input,
+    ) -> Result<crate::compilers::CompilerOutput<Error, Self::CompilerContract>, SolcError> {
+        let solc = self.solc(input)?;
+        let results = self.compile_output::<ResolcInput>(&solc, &input.input)?;
+        let output = std::str::from_utf8(&results).map_err(|_| SolcError::InvalidUtf8)?;
+        let results: ResolcCompilerOutput =
+            serde_json::from_str(output).map_err(|e| SolcError::msg(e.to_string()))?;
+        Ok(results.into())
+    }
+}
+
+impl Resolc {
+    pub fn new(resolc_path: impl Into<PathBuf>, solc_compiler: SolcCompiler) -> Result<Self> {
+        let resolc_path = resolc_path.into();
+        let resolc_version = Self::get_version_for_path(&resolc_path)?;
+        Ok(Self { resolc_version, resolc: resolc_path, solc: solc_compiler })
+    }
+
+    pub fn new_from_path(resolc_path: impl Into<PathBuf>, solc_path: PathBuf) -> Result<Self> {
```

Comment:
`new` and `new_from_path` seem repeat themselves why not init `solc` inside `new_from_path` then call `new` with `solc` as opposed to repeating the logic, We also need to check if the `path` exists inside `new_from_path`

Reply:
removed `new_from_path` completely